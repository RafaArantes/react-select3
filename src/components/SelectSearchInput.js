import React from 'react'
import PropTypes from 'prop-types'
import filterKeyDown from '../utils/filterKeyDown'
import { stopPropagation } from '../utils/events'


const SelectSearchInput = ({ onClick, onKeyDown, ...props }, // eslint-disable-line no-unused-vars
  { cssClassNamePrefix }) => (
  <span className={`${cssClassNamePrefix}__search`}>
    <input
      className={`${cssClassNamePrefix}__search-field`}
      autoFocus
      autoComplete='off'
      autoCorrect='off'
      autoCapitalize='off'
      spellCheck='false'
      onKeyDown={stopPropagation(filterKeyDown([
        {
          allowedKeys: ['Escape', 'Enter', 'ArrowUp', 'ArrowDown'],
          func: onKeyDown,
        },
      ]))}
      onClick={stopPropagation()}
      {...props}
    />
  </span>
);


SelectSearchInput.contextTypes = {
  cssClassNamePrefix: PropTypes.string,
};

export default SelectSearchInput
