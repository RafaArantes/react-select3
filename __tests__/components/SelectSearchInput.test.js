import SelectSearchInput from '../../src/components/SelectSearchInput'


describe('SelectSearchInput', () => {
  it('Should render', () => {
    const component = shallow(
      <SelectSearchInput isPending/>,
      selectComponentContext
    );

    expect(component).toMatchSnapshot()
  })
});
