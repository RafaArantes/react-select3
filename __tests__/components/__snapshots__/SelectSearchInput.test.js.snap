// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectSearchInput Should render 1`] = `
<span
  className="rs3__search"
>
  <input
    autoCapitalize="off"
    autoComplete="off"
    autoCorrect="off"
    autoFocus={true}
    className="rs3__search-field"
    isPending={true}
    onClick={[Function]}
    onKeyDown={[Function]}
    spellCheck="false"
  />
</span>
`;
